webcomponents/tool/src/main/frontend/vue/src/components/avatar.vue

```diff
@@ -0,0 +1,205 @@
+    },
+    getI18nProps: function () {
+      loadProperties("avatar").then((r) => {
+        this.i18n = r;
+      });
+    },
+    checkTheme: function () {
+      this.dark = document.documentElement.classList.contains(
+        "sakaiUserTheme-dark"
+      );
+    },
+  },
+  computed: {
+    variant: function () {
+      if (this.loaded) {
+        return "image";
+      } else {
+        return "text";
+      }
+    },
+    borderRadius: function () {
+      return this.form == "square" ? this.sqareRadius : "50%";
+    },
+    imageStyle: function () {
+      return `
+        border-radius: ${this.borderRadius};
+        ${this.dark ? "filter: brightness(85%)" : ""};
+      `;
+    },
+    fontSize: function () {
+      if (this.text.length <= 3) {
+        return this.size / 2;
+      } else {
+        return this.size / (this.text.length * (2 / 3));
+      }
+    },
+    textStyle: function () {
+      return `
+        border-radius: ${this.borderRadius};
+        ${
+          this.variant == "text"
+            ? `
+          background-color: var(--button-primary-background);
+          color: var(--button-primary-text-color);
+        `
+            : `
+          background-color: var(--sakai-background-color-3);
+          color: var(--sakai-text-color-1);
+        `
+        }
+        width: ${this.size}px;
+        height: ${this.size}px;
+        font-size: ${this.fontSize}px;
+        display: flex;
+        flex-direction: column;
+        justify-content: center;
+        text-align: center;
+      `;
+    },
+    imageUrl: function () {
+      if (!this.userid) return undefined;
+      let url = window.location.protocol + "//" + window.location.host;
+      url += "/direct/profile/";
+      url += this.userid + "/image";
+      url += this.official
+        ? "/official"
+        : this.size <= 80
+        ? "/avatar"
+        : this.size <= 100
+        ? "/thumb"
+        : "/default";
+      if (this.siteid) {
+        url += "?siteid=" + this.siteid;
+      }
+      if (this.cachebusting) {
+        url += "?update=" + this.updateString;
+      }
+      return url;
+    },
+    altText: function () {
+      if (this.username) {
+        return this.i18n.avatar_image_alt_of_user + " " + this.username;
+      } else {
+        return this.i18n.avatar_image_alt_no_user;
+      }
+    },
+  },
+  created: function () {
+    //Listen for a prifile picture change and append a cachebusting string when it happens
+    if (this.cachebusting) {
+      this.forceReload();
+      document.body.addEventListener("avatar-changed", (e) => {
+        this.forceReload();
+      });
+    }
+    //Listen for a theme change. On dark theme img will be dimmed
+    if (!this.nodim) {
+      this.checkTheme();
+      document.body.addEventListener("theme-changed", (e) => {
+        this.checkTheme();
+      });
+    }
+    //Load i18n props
+    this.getI18nProps();
+  },
+};
+</script>
```

Comment:
I have really no idea why you've used vue instead of lit. No idea. It's more verbose and less easy to understand and doesn't map onto vanilla custom elements as clearly as lit-element does.

Reply:
As we were working on the meetings tools UI, we created [reusable components](https://github.com/stetsche/meetings-ui/tree/main/src/components). The first one was the avatar component, which was pretty much done before I started to work on this PR. Having this components created already, brings the opportunity to re-use them in other tools, instead of having repeated similar code with inconsistent results. I think that's the core idea of what we want to achieve with webcomponents, right? Personally find vue easier to understand. Not necessarily just from reading the code, but the amount of resources that are out there really help with the learning process.

Reply:
Why not write it as lit and use it in your Vue tool? All the other components we have are reusable and lit is about the closest to future proof without using the raw w3c spec as you can get. Lit is not hard to understand and we have a load of code in production already.